pkg/querier/querier.go

```diff
@@ -88,25 +90,31 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor) storage.Queryable {
+func NewQueryable(dq, cq storage.Queryable, distributor Distributor, ingesterMaxQueryLookback time.Duration) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
-		dqr, err := dq.Querier(ctx, mint, maxt)
-		if err != nil {
-			return nil, err
-		}
-
 		cqr, err := cq.Querier(ctx, mint, maxt)
 		if err != nil {
 			return nil, err
 		}
 
-		return querier{
-			queriers:    []storage.Querier{dqr, cqr},
+		q := querier{
+			queriers:    []storage.Querier{cqr},
 			distributor: distributor,
 			ctx:         ctx,
 			mint:        mint,
 			maxt:        maxt,
-		}, nil
+		}
+
+		// Include ingester only if maxt is within ingesterMaxQueryLookback w.r.t. current time.
+		if maxt >= time.Now().Add(-ingesterMaxQueryLookback).UnixNano()/1e6 {
```

Comment:
This breaks everything now :) Should be `ingesterMaxQueryLookback == 0 || maxt > ....` no?

Reply:
I wonder why the tests didn't fail. Let me check.

Reply:
Because the test need an update.

Reply:
Nice catch! I have updated.